### PR TITLE
Translation now uses MS Azure Cognitive Services

### PR DESCRIPTION
**Pull request by HilbertGilbertson**
Following the retirement of Microsoft DataMarket at the end of March 2017, the translation command and auto-translate feature will no longer work. Additionally, those new to AB/GAB have not been able to register API keys on DataMarket as this was closed in December 2016.

Given that Microsoft provides an excellent translation API for free (with up to 2 million characters per month), it seems sensible to continue using Microsoft (as opposed to other translation APIs).

The translation command and auto-translate feature are both now fully working in GAB, using the new MS Cognitive Services API.

Existing GAB hosters will need to:

1. obtain a Cognitive Services API key (I will post instructions on how to do so below).
2. edit their Configuration/auth.json to specify the new API key (with the new field: `microsoft_cs_key` [the old microsoft_client_id and microsoft_client_secret fields can be removed]).
3. run `npm update` to install the MsTranslator node module.

### CODE REVIEW

**Comment by HilbertGilbertson** [comment]
## Obtaining an API key for translation

### 1) Create an Azure account [here](https://azure.microsoft.com/en-us/free/)

You will need an existing Microsoft account, or will otherwise be prompted to create one.

#### Will this cost me anything?
TL;DR: **No.**

You will need a valid debit/credit card and a phone number to sign up. This is for Azure's identity verification; you will **not** be charged.

The Azure account will say it's a "_1 month trial_". You can completely ignore this. Signing up for a free account does not hook you on to any kind of paid subscription. After 30 days your account simply converts to a free account (you do not need to cancel anything) and the Translate API can continue to be used with a free account.

### 2) Once created, visit the [Azure Portal](https://portal.azure.com)

- On the left hand side click "More services"
- In the _filter_ field type "cognitive" and select "Cognitive Services accounts"
- Click "Add"
- Enter a name for your API account (e.g. _mybot_)
- Under _API type_ select "**Translator Text API**"
- Under _Pricing Tier_ select "F0 (Up to 2M characters translated per month)"
- Under _Resource group_ choose 'Create new' and give it a name
- Under _API Setting_ you may see an error saying 'Account creation disabled' -> click on it and choose 'Enable' then 'Save'
- Now click 'Create'

image

Under Cognitive Services accounts you should now see your newly-created API. Click on it.
Click on 'Keys'. Here you will see two API keys. You may use either _Key 1_ or _Key 2_; it doesn't matter which.

image

Paste the key into auth.json with the field `microsoft_cs_key` and you're good to go.

**Comment by scarecr0w12** [comment]
Nice fix, works fine on my server!
